cpp/src/hash/hashing.cu

```diff
@@ -70,18 +70,19 @@ std::unique_ptr<column> serial_murmur_hash3_32(table_view const& input,
     output_view.begin<int32_t>(),
     output_view.end<int32_t>(),
     [device_input = *device_input, nulls = has_nulls(leaf_table), seed] __device__(auto row_index) {
-      return thrust::reduce(thrust::seq,
-                            device_input.begin(),
-                            device_input.end(),
-                            seed,
-                            [rindex = row_index, nulls] __device__(auto hash, auto column) {
-                              return cudf::type_dispatcher(
-                                column.type(),
-                                element_hasher_with_seed<hash_function, nullate::DYNAMIC>{
-                                  nullate::DYNAMIC{nulls}, hash, hash},
-                                column,
-                                rindex);
-                            });
+      return thrust::reduce(
```

Comment:
We wrote the device-compatible `accumulate` to avoid calling `thrust::reduce` with non-commutative, non-associative operators. Here, it appears that we are using `thrust::reduce` with a non-commutative, non-associative operator. (Even with the `thrust::seq` execution policy, we should avoid relying on an implementation detail.)

This may be best for a follow-up PR. I'll assess this at the end of my review.